src/redirect.rs

```diff
@@ -179,6 +182,16 @@ pub fn check_redirect(policy: &RedirectPolicy, next: &Url, previous: &[Url]) ->
     }).inner
 }
 
+pub fn remove_sensitive_headers(mut headers: Headers, next: &Url, previous: &[Url]) -> Headers {
```

Comment:
It's probably more idiomatic to take a mutable reference of the headers, `&mut Headers`, instead of returning them.